nixpkgs/github.json

```diff
@@ -1,7 +1,7 @@
 {
   "url": "https://github.com/NixOS/nixpkgs",
-  "rev": "0728c3e026e3ac2542a9673d57533829e552e218",
-  "date": "2019-05-17T15:35:26+02:00",
-  "sha256": "0sp8giack8na9hdm8ivk969mpwfcavqrfy7kncx0a1hs1ykw15hy",
+  "rev": "538639a55ce915effc543b3e9c6c901f8ea624c1",
```

Comment:
Looks like the latest commit on `channels/nixos-19.03-small`. I would prefer `channels/nixos-19.03` because small doesn't appear to have Haskell stuff in it, but this is probably fine.

Reply:
Are we safe to go to 19.03? Aren't our mac builders still unable to cope with it?

Reply:
The nixpkgs pin is just for running the tests in Haskell.nix.

iohk-nix will provide its own nixpkgs - our iohk fork - and pass it to haskell.nix.

Reply:
The Hydra jobset for Haskell.nix is only for Linux at the moment. I would like to also test on MacOS, but don't want to overload our mac builders because the tests tend to rebuild quite a lot.

Reply:
:+1:

Ideally we would test using the same nixpkgs that we're going to use later, but it probably doesn't matter for these tests.

Reply:
@rvl hmm, it's been

```
nix-prefetch-git https://github.com/nixos/nixpkgs refs/heads/release-19.03 > nixpkgs/github.json
```

Reply:
https://github.com/NixOS/nixpkgs/tree/538639a55ce915effc543b3e9c6c901f8ea624c1

Hmm what haskell stuff do you think is missing?

Reply:
@michaelpj I think 19.03 is the right version for Haskell.nix to test against, because "works with the latest stable NixOS release" is what users in general would expect. Ideally iohk-nix would have tests, and these could include running the Haskell.nix tests against our input-output-hk/nixpkgs fork, good idea.

@angerman My understanding is that the `release-19.03` branch is not a channel. So it _could_ have a broken GHC (although that's unlikely). The `nixos-19.03-small` channel will advance when a select number of packages pass in Hydra (haskell not included), then the `nixos-19.03` channel will advance once all the NixOS tests build and pass in Hydra. So the correct prefetch would be:
```
nix-prefetch-git https://github.com/NixOS/nixpkgs-channels refs/heads/nixos-19.03 > nixpkgs/github.json
```

Maybe we could make that the policy and document it?

Reply:
@rvl great idea with the policy and documentation. I just wonder if there is a way we can enforce this? Maybe we can have buildkite check that the rev is part of the channel? And if not complain with a message how to update?

Reply:
I've been hoping we could move to [niv](https://github.com/nmattia/niv) for managing our pins (I think it's not quite ready yet). The pin specifications for `niv` include the tracking branch, so this sort of thing is easier to get right.